doc/classes/vector.adoc

```diff
@@ -0,0 +1,260 @@
+= Vector API
+include::../../include/config.adoc[]
+:description: Documentation for the Vector class and API
+:keywords: mod, library, vector, x, y, z, pos, position
+
+TIP: The respective source code is located https://github.com/minetest/minetest/blob/master/builtin/common/vector.lua[here].
+
+== `vector` Namespace
+
+NOTE: The `vector` type used to be a simple `table` with `x`, `y`, and `z` values.
+However it has more recently been given metatable methods for convenience.
+Unless otherwise noted, the `vector` type always refers to the metatable-enhanced variety.
+Due note that functions here will accept an old-style (non-metatable) `vector`, but you cannot perform metatable operations with said `vector`.
+
+=== `vector.new(a, b, c) -> vector`
+* `a`: `number`, `vector`, or `nil`
+* `b`, `c`: `number` or `nil`
+
+If `a`, `b`, and `c` are `number`:: Returns a new `vector` where `{x = a, y = b, z = c}`
+If `a` is a `vector`:: Returns `vector.copy(a)`
+If all parameters are `nil`:: Returns `vector.zero()`
+
+=== `vector.zero() -> vector`
+Returns a new `vector` where `{x = 0, y = 0, z = 0}`
+
+=== `vector.copy(v) -> vector`
+* `v`: `vector`
+
+Returns a new `vector` where `{x = v.x, y = v.y, z = v.z}`
+
+=== `vector.from_string(s, init) -> vector, string`
```

Comment:
```suggestion
=== `vector.from_string(s, init) -> vector, number`
```

Reply:
I'm an idiot, I saw the last pair of parentheses in the capture string and thought _that_ was `np` and the name was just stupid, not realizing that they were empty and thus wouldn't capture anything.